devlog/_plan/260828_cursor_ndjson_backlog_train/031_midstream_echo_fix.md

```
@@ -0,0 +1,96 @@
+  (b) emits. BOTH release paths route through it: releaseGuardHeld()'s
+  per-held-event emit for text deltas, and the ordinary post-guard emit at
+  cursor.ts:~324. Held deltas are NOT fed while held — only on release —
+  so no double-feed is possible.
+- At turn end (done event handling, before final emit): read
+  observer.findings(); for each finding emit debugProviderDiagnostic
+  ("cursor", "midstream-envelope-echo", { wireModel, conversationHash:
+  request.conversationId.slice(0,16), offset, marker, callIdCorrupt }).
+  marker stays a fixed enum string; no content bytes logged (audit
+  finding 6 conventions).
+
+### 3. MODIFY tests/cursor-envelope-echo-retry.test.ts (named activation
```

markdownlint-cli2 (0.23.2)

[warning] Headings should be surrounded by blank lines
Expected: 1; Actual: 0; Below

(MD022, blanks-around-headings)